### PR TITLE
Add support for Lambda

### PR DESCRIPTION
**Pull request by nickcaballero**
This PR adds support for invoking Lambda functions.

### CODE REVIEW

File: tests/internal/helpers.js [hunk 1]
```diff
@@ -50,7 +51,16 @@ import { describe, expect } from 'https://jslib.k6.io/k6chaijs/4.3.4.1/index.js'
...
+            return expect(what, description)
+        })
+    } catch (error) {
+        if (error.name !== 'AssertionError') {
+            console.error(`FAIL [${description}]`, error.stack);
+            check(error, {[description]: () => false})
+        } else {
+            console.error(`FAIL [${description}]`, error.message);
+        }
+    }
```
Comment (nickcaballero) [inline tests/internal/helpers.js:65]: Adding this bit to provide better error messages. While testing I noticed that we weren't really adding a failed check when the test failed without any failed assertions.

File: src/internal/error.ts [hunk 2]
```diff
@@ -35,4 +36,12 @@ export class AWSError extends Error {
...
         return new AWSError(doc.find('Message').text(), doc.find('Code').text())
     }
+
+    static parse(response: Response): AWSError {
+        if (response.headers['Content-Type'] === 'application/json') {
+            return new AWSError(response.json('message') as string, response.headers['X-Amzn-Errortype'])
+        } else {
+            return AWSError.parseXML(response.body as string);
+        }
+    }
```
Comment (nickcaballero) [inline src/internal/error.ts:46]: Updating a more generic parsing method since responses are not always in XML.

**Review by codebien — COMMENTED** [review]
Hey @nickcaballero,
thanks for this new contribution! :bow:

We already have a similar PR https://github.com/grafana/k6-jslib-aws/pull/71 in the pipeline regarding the Lambda service. Our plan would be to merge https://github.com/grafana/k6-jslib-aws/pull/71 before then it would be great to have this PR rebased on top of it.
I see you have added additional details and a different proposal for the `invoke` API that we would like to include in the project.

Does it sound like a reasonable plan for you?

**Comment by nickcaballero** [comment]
> Hey @nickcaballero, thanks for this new contribution! 🙇
> 
> We already have a similar PR #71 in the pipeline regarding the Lambda service. Our plan would be to merge #71 before then it would be great to have this PR rebased on top of it. I see you have added additional details and a different proposal for the `invoke` API that we would like to include in the project.
> 
> Does it sound like a reasonable plan for you?

Sounds good! Apologies, I didn't see the other PR 🤦.

**Comment by codebien** [comment]
Hey @nickcaballero, we merged #71. You can now rebase on top of it. Thanks for the effort :bow:

File: src/internal/error.ts [hunk 3]
```diff
@@ -35,4 +36,15 @@ export class AWSError extends Error {
         const doc = parseHTML(xmlDocument)
         return new AWSError(doc.find('Message').text(), doc.find('Code').text())
     }
+
+    static parse(response: Response): AWSError {
+        if (response.headers['Content-Type'] === 'application/json') {
+            const error = response.json() as any;
+            const message = error.Message || error.message || error.__type || 'An error occurred on the server side';
+            const code = response.headers['X-Amzn-Errortype'] || error.__type
+            return new AWSError(message, code)
```
Comment (nickcaballero) [inline src/internal/error.ts:45]: Pulled some of the error handling logic that @jakub-qg into here to make it more reusable.

**Comment by nickcaballero** [comment]
> Hey @nickcaballero, we merged #71. You can now rebase on top of it. Thanks for the effort 🙇

Done!

File: README.md [hunk 4]
```diff
@@ -375,6 +376,34 @@ export default async function () {
 }
 ```
 
+### Lambda
+
+Consult the `LambdaClient` [dedicated k6 documentation page](https://k6.io/docs/javascript-api/jslib/aws/lambdaclient) for more details on its methods and how to use it.
+
+```javascript
+import { AWSConfig, LambdaClient } from 'https://jslib.k6.io/aws/0.11.0/lambda.js'
```
Comment (codebien) [inline README.md:384]:
```suggestion
import { AWSConfig, LambdaClient } from 'https://jslib.k6.io/aws/0.12.0/lambda.js'
```
I expect we need to bump, right @oleiade?

File: localstack/init/ready.d/lambda.sh [hunk 5]
```diff
@@ -1,21 +1,18 @@
...
-# Create a dummy lambda function responding with a static string "Hello World!"
-cat >index.js <<EOF
-exports.handler = async function(event, context) {
-    return "Hello World!";
-}
-EOF
+for file in "$testdata_folder"/*; do
+  function_name=$(basename "$file")
+  function_zip="$zip_dir/$function_name.zip"
+  (cd "$file" || exit; zip "$function_zip" ./*)
```
Comment (codebien) [inline localstack/init/ready.d/lambda.sh:10]: @nickcaballero do we need to have them nested in folders? Could we have direct a list of files?

Comment (nickcaballero) [reply localstack/init/ready.d/lambda.sh:10]: The file structure matches how the zips are created. We can flatten the folders, but it'll make the script a bit more complicated.

File: README.md (hunk 4 above)
Comment (allansson) [reply README.md:384]: Yes, we should bump it. Some properties have been changed in a way that breaks the contract. The end result is more idiomatic JS and a good change, but on the off chance that someone started using the current version we should avoid breaking their code.

File: localstack/init/ready.d/lambda.sh (hunk 5 above)
Comment (oleiade) [reply localstack/init/ready.d/lambda.sh:10]: I would prefer to have a simpler script, rather than a simpler folder structure 👍🏻

**Review by codebien — APPROVED** [review]

**Pull request merged**